src/Perl6/Metamodel/EnumHOW.nqp

```diff
@@ -85,13 +88,17 @@ class Perl6::Metamodel::EnumHOW
     }
 
     method enum_from_value($obj, $value) {
-        unless %!value_to_enum {
+        unless $!value_to_enum {
+            my $value_to_enum := nqp::hash;
             for @!enum_value_list {
-                %!value_to_enum{$_.value} := $_;
+                $value_to_enum{$_.value} := $_;
             }
+            nqp::scwbdisable();
+            $!value_to_enum := $value_to_enum;
+            nqp::scwbenable();
         }
-        nqp::existskey(%!value_to_enum, $value)
-            ?? %!value_to_enum{$value}
+        nqp::existskey($!value_to_enum, $value)
+            ?? $!value_to_enum{$value}
```

Comment:
Ah, I should have added: it's important only read the `$!value_to_enum` attribute once into a lexical, not multiple times, otherwise it could have `NQPMu` assigned along the way. That won't make a SEGV, but it would make an exception.